src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -598,17 +600,19 @@ public Object sliceCapture(
                 DynamicObject regexp,
                 Object capture,
                 @Cached("createOnSelf()") CallDispatchHeadNode callNode,
-                @Cached("create()") RegexpSetLastMatchPrimitiveNode setLastMatchNode) {
+                @Cached("createOnSelf()") CallDispatchHeadNode setLastMatchNode,
+                @Cached("create()") ReadCallerFrameNode readCallerNode) {
             final Object matchStrPair = callNode.call(frame, string, "subpattern", regexp, capture);
 
+            DynamicObject binding = BindingNodes.createBinding(getContext(), readCallerNode.execute(frame).materialize());
             if (matchStrPair == nil()) {
-                setLastMatchNode.executeSetLastMatch(frame, nil());
+                setLastMatchNode.call(frame, getContext().getCoreLibrary().getTruffleRegexpOperationsModule(), "set_last_match", nil(), binding);
                 return nil();
             }
 
             final Object[] array = (Object[]) Layouts.ARRAY.getStore((DynamicObject) matchStrPair);
 
-            setLastMatchNode.executeSetLastMatch(frame, array[0]);
+            setLastMatchNode.call(frame, getContext().getCoreLibrary().getTruffleRegexpOperationsModule(), "set_last_match", array[0], binding);
```

Comment:
Can you use the `coreLibrary()` helper? It'd gain quite a bit of width on this long line (and above).